src/main/java/ch/njol/skript/expressions/ExprSpawnerType.java

```diff
@@ -39,55 +22,93 @@
 @Description("Retrieves, sets, or resets the spawner's entity type")
-@Examples({"on right click:",
-		"	if event-block is spawner:",
-		"		send \"Spawner's type is %target block's entity type%\""})
-@Since("2.4")
+@Examples({
+	"on right click:",
+	"\tif event-block is spawner:",
+	"\t\tsend \"Spawner's type is %target block's entity type%\""
+})
+@Since("2.4, INSERT VERSION (trial spawner)")
 public class ExprSpawnerType extends SimplePropertyExpression<Block, EntityData> {
-	
+
+	private static final boolean HAS_TRIAL_SPAWNER = Skript.classExists("org.bukkit.block.TrialSpawner");
+
 	static {
 		register(ExprSpawnerType.class, EntityData.class, "(spawner|entity|creature) type[s]", "blocks");
 	}
-	
-	@Override
+
 	@Nullable
 	public EntityData convert(Block block) {
-		if (!(block.getState() instanceof CreatureSpawner))
-			return null;
-		EntityType type = ((CreatureSpawner) block.getState()).getSpawnedType();
-		if (type == null)
-			return null;
-		return EntityUtils.toSkriptEntityData(type);
+		if (block.getState() instanceof CreatureSpawner) {
+			EntityType type = ((CreatureSpawner) block.getState()).getSpawnedType();
+			if (type == null)
+				return null;
+			return EntityUtils.toSkriptEntityData(type);
+		}
+		if (HAS_TRIAL_SPAWNER && block.getState() instanceof TrialSpawner) {
+			TrialSpawner trialSpawner = (TrialSpawner) block.getState();
+			EntityType type;
+			if (trialSpawner.isOminous()) {
+				type = trialSpawner.getOminousConfiguration().getSpawnedType();
+			} else {
+				type = trialSpawner.getNormalConfiguration().getSpawnedType();
+			}
+			if (type == null)
+				return null;
+			return EntityUtils.toSkriptEntityData(type);
+		}
+		return null;
 	}
 	
 	@Nullable
 	@Override
 	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.RESET) 
-			return CollectionUtils.array(EntityData.class);
+		switch (mode) { // needs update next version
+			case SET:
+			case RESET:
+				return CollectionUtils.array(EntityData.class);
+			}
 		return null;
 	}
 	
 	@SuppressWarnings("null")
 	@Override
 	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
-		for (Block b : getExpr().getArray(event)) {
-			if (!(b.getState() instanceof CreatureSpawner))
-				continue;
-			CreatureSpawner s = (CreatureSpawner) b.getState();
-			switch (mode) {
-				case SET:
-					assert delta != null;
-					s.setSpawnedType(EntityUtils.toBukkitEntityType((EntityData) delta[0]));
-					break;
-				case RESET:
-					s.setSpawnedType(org.bukkit.entity.EntityType.PIG);
-					break;
+		for (Block block : getExpr().getArray(event)) {
+			if (block.getState() instanceof CreatureSpawner) {
+				CreatureSpawner spawner = (CreatureSpawner) block.getState();
+				switch (mode) {
+					case SET:
+						assert delta != null;
+						spawner.setSpawnedType(EntityUtils.toBukkitEntityType((EntityData) delta[0]));
+						break;
+					case RESET:
+						spawner.setSpawnedType(EntityType.PIG);
```

Comment:
This should probably be changed to null, though I'd like other opinions on whether doing that is valid for a patch change

Reply:
i don't think it should, well, atleast not for this PR, since it would be a breaking logic change